src/speedrun.c

```diff
@@ -235,3 +239,96 @@ void speedrun_map_completed()
     // closes the connection
     mysql_close(con);
 }
+
+
+struct string {
+  char *ptr;
+  size_t len;
+};
+
+void init_string(struct string *s) {
+  s->len = 0;
+  s->ptr = malloc(s->len+1);
+  if (s->ptr == NULL) {
+    fprintf(stderr, "malloc() failed\n");
+    exit(EXIT_FAILURE);
+  }
+  s->ptr[0] = '\0';
+}
+
+size_t url_result_write(void *ptr, size_t size, size_t nmemb, struct string *s)
+{
+  size_t new_len = s->len + size*nmemb;
+  s->ptr = realloc(s->ptr, new_len+1);
+  if (s->ptr == NULL) {
+    fprintf(stderr, "realloc() failed\n");
+    exit(EXIT_FAILURE);
+  }
+  memcpy(s->ptr+s->len, ptr, size*nmemb);
+  s->ptr[new_len] = '\0';
+  s->len = new_len;
+
+  return size*nmemb;
+}
+
+// yeah idk what's written in here... 
+// but, as some reposter on r/ProgrammerHumor said: "If it works, don't touch it"
+void send_best_time()
+{
+    char url[67];
+    int mapnum = gamemap-1;
+    snprintf(url, 67, BEST_SCORE_ON_MAP_URL, mapnum);
+
+
+    CURL *curl;
+    int result;
+
+    curl = curl_easy_init();
+
+    struct string retrieved_data;
+    init_string(&retrieved_data);
+    
+    curl_easy_setopt(curl, CURLOPT_URL, url);
+    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, url_result_write);
+    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &retrieved_data);
+    curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
```

Comment:
Is this necessary? Does verification not work?

Reply:
Oh no, I need to skip verification, my python code I wrote a while ago on my phone needed to skip verification too

Reply:
All right then